iree/compiler/Dialect/Stream/Transforms/ScheduleAllocation.cpp

```diff
@@ -0,0 +1,1440 @@
+        }
+      }
+    }
+  }
+
+  // Inverst the value aliaser->aliasee map so that we have for any particular
+  // value the list of all other values that alias it.
+  for (auto it : valueAliases) {
+    for (auto aliasee : it.second) {
+      for (auto aliaser : it.second) {
+        if (aliaser != aliasee) {
+          valueAliases[aliasee].insert(aliaser);
+        }
+      }
+    }
+  }
+
+  return valueAliases;
+}
+
+//===----------------------------------------------------------------------===//
+// Liveness interval analysis
+//===----------------------------------------------------------------------===//
+
+static constexpr int LIVE_IN = INT_MIN;
+static constexpr int LIVE_OUT = INT_MAX;
+struct LivenessInterval {
+  int start = 0;
+  int end = 0;
+  int ordinal = -1;  // unique per value
+  Value value;
+  bool operator<(const LivenessInterval &rhs) const {
+    return ordinal < rhs.ordinal;
+  }
+};
+using LivenessIntervalMap = DenseMap<Value, LivenessInterval>;
+using LivenessIntervalList = SmallVector<LivenessInterval>;
+
+// Computes the liveness intervals for each value in the execution region.
+// Returns a closed range over an arbitrary operation ordering. The LIVE_IN and
+// LIVE_OUT sentinels will be used to indicate values that are live-in and
+// live-out to the execution region (captured input arguments and escaping
+// output results).
+//
+// All values will have a range with aliased values sharing the union of their
+// constituent ranges - including block arguments. Note that not all values will
+// have buffers allocated to them - we are just tracking transitive SSA value
+// lifetime.
+static LivenessIntervalList computeExecutionRegionLivenessIntervals(
+    IREE::Stream::AsyncExecuteOp executeOp,
+    const ValueAliasingMap &valueAliases) {
+  // Perform a liveness analysis on the execution region.
+  // Fragments have a single block and as such the live-in/live-out block
+  // information derived here applies to the entire stream region.
+  assert(executeOp.body().getBlocks().size() == 1);
+  auto *streamBlock = &executeOp.body().front();
+  Liveness streamLiveness(executeOp);
+  auto *livenessInfo = streamLiveness.getLiveness(streamBlock);
+
+  // Operations don't allow us to get their already computed order so we make up
+  // our own. We have a single block and thus the ordering is complete.
+  DenseMap<Operation *, int> opOrdering;
+  for (auto &op : *streamBlock) {
+    opOrdering[&op] = opOrdering.size();
+  }
+
+  // Liveness doesn't track return values as live-outs so we do that here.
+  SmallPtrSet<Value, 16> liveOuts;
+  auto yieldOp = cast<IREE::Stream::YieldOp>(streamBlock->back());
+  for (auto returnValue : yieldOp.operands()) {
+    if (!returnValue.getType().isa<IREE::Stream::ResourceType>()) continue;
+    liveOuts.insert(returnValue);
+  }
+
+  // Compute live-in intervals special as we won't catch them in the op walk
+  // below as they are block arguments.
```

Comment:
nit: using "special" as an adverb? This is confusing

```suggestion
  // Compute live-in intervals by hand as we won't catch them in the op walk
  // below since they are block arguments.
```